zaza/openstack/charm_tests/test_utils.py

```diff
@@ -154,6 +154,36 @@ def setUpClass(cls, application_name=None, model_alias=None):
             model_name=cls.model_name)
         logging.debug('Leader unit is {}'.format(cls.lead_unit))
 
+    def config_current_separate_special_keys(
+            self, special_type_keys, config_keys=None,
+            application_name=None):
+        """Obtain current config and the special type config separately.
+
+        :param config_keys: iterable of strs to index into the current config.
+                            If None, return all keys from the config
+        :type config_keys:  Optional[Iterable[str]]
+        :param special_type_keys: list of keys to be separated out
+        :type special_type_keys: list
+        :param application_name: String application name for use when called
+                                 by a charm under test other than the object's
+                                 application.
+        :type application_name:  Optional[str]
+        :return: Dictionary of previous charm configs applied without the
```

Comment:
I also don't understand this (I swear I read it five times and I was very focused). What does it mean to "apply charm configs" and what are "previous" configs and what does it mean to apply them "without the special type keys"?

Reply:
I mean if it was a "config_changed" context manager I would understand concepts like "applying" or "previous" at least but isn't it just a config getter?